### PR TITLE
Issue 1775: Upgrade to Cassandra to 4.0-beta2

### LINKED ISSUE #1775: Upgrade to Cassandra to 4.0-alpha4 or higher
# Task Description

We should consider upgrading to `org.apache.cassandra:cassandra-all:4.0-alpha4`, as it can help us with our performance issues.

# Tasks

The following tasks will need to be carried out:
* [ ] Create a branch off `issues/1649` against strongbox/strongbox-parent and bump the version.
* [ ] Sort out the issues that pop up.
* [ ] Check how it affects the build times of `strongbox-against-strongbox`.
* [ ] Investigate how [the vulnerabilities reported by Snyk.io against this version of `cassandra-all`[(https://snyk.io/vuln/maven:org.apache.cassandra:cassandra-all) could be fixed.

# Task Relationships

This task:
* Is a sub-task of: #1649

# Useful Links

* [Cassandra 4.0-alpha4: Benchmarking streaming](https://cassandra.apache.org/blog/2019/04/09/benchmarking_streaming.html)

# Help

* [Our chat](https://chat.carlspring.org/)
* Points of contact:
  * @carlspring
  * @sbespalov
  * @steve-todorov

### PR DESCRIPTION
**Pull request by sbespalov**
# Pull Request Description

This pull request closes #1775 

# Acceptance Test

* [ ] Building the code with `mvn clean install -Dintegration.tests` still works.
* [ ] Running `mvn spring-boot:run` in the `strongbox-web-core` still starts up the application correctly.
* [ ] Building the code and running the `strongbox-distribution` from a `zip` or `tar.gz` still works.
* [ ] The tests in the [`strongbox-web-integration-tests`](https://github.com/strongbox/strongbox-web-integration-tests/) still run properly.

# Questions

* Does this pull request break backward compatibility? 
  * [ ] Yes
  * [ ] No

* Does this pull request require other pull requests to be merged first? 
  * [ ] Yes, please see #...
  * [ ] No

* Does this require an update of the documentation?
  * [ ] Yes, please see strongbox/strongbox-docs#{PR_NUMBER}
  * [ ] No

# Code Review And Pre-Merge Checklist

* [ ] My code follows the [coding convention](https://strongbox.github.io/developer-guide/coding-convention.html) of this project.
* [ ] I have performed a self-review of my own code.
* [ ] I have commented my code in hard-to-understand areas.
* [ ] My changes generate no new warnings.
* [ ] I have added tests that prove my fix is effective or that my feature works.
* [ ] New and existing unit tests pass locally with my changes.

### CODE REVIEW

**Review by carlspring — CHANGES_REQUESTED** [review]
@sbespalov ,

Why is there so much noise in this pull request? You seem to have re-written the history in a way that all of the changes in from the base branch (`issues/1649`) seem to be "new" changes, when in fact the branch for the jdk11 changes should already be based off the `issues/1649`. With all this noise it's hard to tell what the actual changes for the upgrade really were. Would it be possible to sort this out? Thanks! :)

File: strongbox-cron/strongbox-cron-tasks/pom.xml [hunk 1]
```diff
@@ -163,12 +159,10 @@
...
         <dependency>
-            <groupId>org.apache.directory.server</groupId>
-            <artifactId>apacheds-core</artifactId>
-        </dependency>
-        <dependency>
-            <groupId>org.apache.directory.server</groupId>
-            <artifactId>apacheds-protocol-ldap</artifactId>
+            <groupId>com.unboundid</groupId>
+            <artifactId>unboundid-ldapsdk</artifactId>
+            <version>5.0.1</version>
```
Comment (carlspring) [inline strongbox-cron/strongbox-cron-tasks/pom.xml:164]: This needs to go into the `strongbox-parent`.

Comment (sbespalov) [reply strongbox-cron/strongbox-cron-tasks/pom.xml:164]: fixed

**Comment by sbespalov** [comment]
@carlspring the conflicts is because of `issues/1649` was rebased and it affected on this branch because it was based on old `issues/1649` commits

File: strongbox-cron/strongbox-cron-tasks/pom.xml [hunk 2]
```diff
@@ -7,13 +7,13 @@
     <parent>
         <groupId>org.carlspring.strongbox</groupId>
         <artifactId>strongbox-parent</artifactId>
-        <version>1.0-PR-62-SNAPSHOT</version>
+		<version>1.0-issues-1775-SNAPSHOT</version>
```
Comment (carlspring) [inline strongbox-cron/strongbox-cron-tasks/pom.xml:10]:
```suggestion
        <version>1.0-issues-1775-SNAPSHOT</version>
```

File: strongbox-resources/strongbox-common-resources/src/main/resources/logback-spring.xml [hunk 3]
```diff
@@ -114,7 +106,7 @@
         <logger name="org.carlspring.strongbox.data.server" level="DEBUG"/>
         <!-- Strongbox packages -->
 
-        <!-- Third-party packages -->
+        <!-- Third-party packages -->
```
Comment (carlspring) [inline strongbox-resources/strongbox-common-resources/src/main/resources/logback-spring.xml:109]:
```suggestion
        <!-- Third-party packages -->
```

File: strongbox-security/strongbox-authentication-registry/pom.xml [hunk 4]
```diff
@@ -8,12 +8,12 @@
...
         <groupId>org.carlspring.strongbox</groupId>
         <artifactId>strongbox-parent</artifactId>
-        <version>1.0-PR-62-SNAPSHOT</version>
+        <version>1.0-issues-1775-SNAPSHOT</version>
         <relativePath/>
     </parent>
 
     <artifactId>strongbox-authentication-registry</artifactId>
-    <version>1.0-PR-62-SNAPSHOT</version>
+	<version>1.0-issues-1775_new-SNAPSHOT</version>
```
Comment (carlspring) [inline strongbox-security/strongbox-authentication-registry/pom.xml:16]:
```suggestion
    <version>1.0-issues-1775_new-SNAPSHOT</version>
```

**Review by carlspring — APPROVED** [review]
LGTM! :)

**Pull request merged**